### PR TITLE
AWS API Gateway customize log level

### PR DESCRIPTION
**Pull request by ktardif**
## What did you implement:

Added support for specifying the AWS API Gateway log level to be used. I also updated the sample AWS serverless.yml.md file to include the log format option as that was missing when that feature was added.

This is an enhancement to API Gateway logs as part of #6094.

## How did you implement it:

Users can now customize the log level by setting the `provider.logs.restApi.level` optional setting in serverless.yml to either INFO (default value) or ERROR:

```yml
provider:
  name: aws
  logs:
    restApi: 
      level: ERROR
```

An error will be thrown if something other than INFO or ERROR is provided. I updated updateStage.js to use the user-provided log level if available, otherwise it falls back to the default INFO as before.

## How can we verify it:

Create a serverless project and set `provider.logs.restApi.level` to either INFO or ERROR and run a deploy. Set to something to cause an invalid value error.

## Todos:

- [x] Write tests and confirm existing functionality is not broken.  
- [x] Write documentation
- [x] Ensure there are no lint errors.  
- [x] Ensure introduced changes match Prettier formatting.  
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js [hunk 1]
```diff
@@ -16,13 +16,17 @@ const defaultApiGatewayLogFormat = [
   'protocol: $context.protocol',
   'responseLength: $context.responseLength',
 ].join(', ');
+const defaultApiGatewayLogLevel = 'INFO';
+const apiGatewayValidLogLevels = ['INFO', 'ERROR'];
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js]: nit: conceptually it'll be nicer to use `Set` here, still then stringification for error message requires to resolve string as `Array.from(apiGatewayValidLogLevels).join(', ')

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js [hunk 2]
```diff
@@ -191,6 +195,18 @@ function handleLogs() {
       logFormat = logs.format;
     }
 
+    let level = defaultApiGatewayLogLevel;
+    if (logs.level) {
+      level = logs.level;
+      if (!apiGatewayValidLogLevels.includes(level)) {
+        throw new Error(
```
[the remark this answers is not in the document]
Comment (ktardif) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js]: Thanks, I had looked in the same file and saw Error was used so I had just used that.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js (hunk 1 above)
Comment (ktardif) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js]: Agreed, it's semantically more correct; I've switched it over.

File: lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (medikoo) [reply lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js]: Yes, we've just recently (with #6502) introduced a distinction of how we communicate the _user_ and _programmer_ errros_, and before that it was not perfectly respected

**Pull request merged**